pkgs/by-name/wh/whatsapp-for-mac/package.nix

```diff
@@ -0,0 +1,57 @@
+{ lib
+, fetchzip
+, stdenvNoCC
+, writeShellApplication
+, curl
+, xmlstarlet
+, common-updater-scripts
+}:
+
+stdenvNoCC.mkDerivation (finalAttrs: {
+  pname = "whatsapp-for-mac";
+  version = "2.24.11.85";
+
+  src = fetchzip {
+    extension = "zip";
+    name = "WhatsApp.app";
+    url = "https://web.whatsapp.com/desktop/mac_native/release/?version=${finalAttrs.version}&extension=zip&configuration=Release&branch=relbranch";
+    hash = "sha256-vbgym7AfRi6kzZVMPRA6lP+6pL8cXnR7yt9hWPnt3+E=";
+  };
+
+  dontConfigure = true;
+  dontBuild = true;
+
+  sourceRoot = ".";
```

Comment:
Is there a need to manually set `sourceRoot` here?

Reply:
Yes otherwise:

```console
@nix { "action": "setPhase", "phase": "unpackPhase" }
Running phase: unpackPhase
unpacking source archive /nix/store/p6if257110mplhs4rk6f0gm3vgf108f3-WhatsApp.app
source root is WhatsApp.app
@nix { "action": "setPhase", "phase": "patchPhase" }
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "installPhase" }
Running phase: installPhase
cp: missing destination file operand after '/nix/store/vchh2i1dz0n6pnl4hyby98ld312ayih2-whatsapp-for-mac-2.24.11.85/Applications'
Try 'cp --help' for more information.
```